src/GovUk.Education.ExploreEducationStatistics.Data.Model/Migrations/Current_Routine_RebuildIndexes.sql

```diff
@@ -1,120 +1,302 @@
-                    IF (@CurrentTable != @LastTable)
-                        BEGIN
-                            -- Update all statistics on the table
-                            SET @Command = 'UPDATE STATISTICS ' + @CurrentTable;
-                            RAISERROR ('Updating statistics on table ''%s''.', 0, 1, @CurrentTable) WITH NOWAIT;
-                            EXEC (@Command);
+                    -- We set the StartTime even when ActionRequired = 'None'. This allows us to update rows that
+                    -- won't be processed if we've exceeded @StopTime
+                    UPDATE __Log_RebuildIndexesAlterIndexes
+                    SET HitTimeout = 1
+                    WHERE StartTime IS NULL;
 
-                            SET @LastTable = @CurrentTable;
-                        END
+                    SET @SkipRemainingTasks = 1;
+                END
 
-                    SET @StatsRowIndex += 1;
+            SET @StatsRowIndex += 1;
+        END
+
+    RAISERROR ('Completed optimising indexes for tables.', 0, 1, NULL) WITH NOWAIT;
+
+    /*-----------------------------------------------------------------------*/
+
+    -- Call UPDATE STATISTICS on any tables that have had their indexes rebuilt/reorganized
+    SET @TablesRowIndex = 1;
+    SELECT @TablesCount = COUNT(Id) FROM #TableList;
+    WHILE @TablesRowIndex <= @TablesCount AND @SkipRemainingTasks = 0
+        BEGIN
+            SELECT @CurrentTable = ObjectName
+            FROM #TableList
+            WHERE Id = @TablesRowIndex;
+
+            SET @Command = 'UPDATE STATISTICS ' + @CurrentTable;
+
+            -- Save start time
+            UPDATE __Log_RebuildIndexesUpdateStatistics
+            SET StartTime = GETUTCDATE()
+            WHERE RunId = @RunId
+              AND TableName = @CurrentTable;
+
+            -- Skip if no indexes related to table were rebuilt/reorganized
+            SELECT @CurrentTableIndexesProcessed = COUNT(*)
+            FROM __Log_RebuildIndexesAlterIndexes
+            WHERE RunId = @RunId
+              AND ObjectName = @CurrentTable
+              AND EndTime IS NOT NULL
+
+            IF @CurrentTableIndexesProcessed = 0
+                BEGIN
+                    RAISERROR ('No indexes related to table ''%s'' updated, so skipping update statistics.',
+                        0, 1, @CurrentTable) WITH NOWAIT;
+                    SET @TablesRowIndex += 1;
+                    CONTINUE;
                 END
-            IF(@StopTime > GETUTCDATE())
-                RAISERROR ('Completed updating statistics on tables.', 0, 1, NULL) WITH NOWAIT;
-            ELSE
-                RAISERROR ('Reindexing did not complete in %d minutes. Remaining reindexing skipped.', 16, 1, @StopInMinutes) WITH NOWAIT;
+
+            -- Run
+            RAISERROR ('Updating statistics on table ''%s''.', 0, 1, @CurrentTable) WITH NOWAIT;
+            EXEC (@Command);
```

Comment:
I would move the assignment of `@Command` directly before this so that it's set at the point that it's needed. If you go back a few revisions it was like this previously.